### PR TITLE
add merge workflow between load raw api model and validation

### LINKED ISSUE #1269: Validation failed on scale command
**Is this a request for help?**: NO

---

**Is this an ISSUE or FEATURE REQUEST?** (choose one):
ISSUE
---

**What version of acs-engine?**:
all the versions

---

Currently, there validation of user's request to scale always failed. Because, when user scale, the servicePrincipalProfile.Secret is left empty. The validation need it not empty.

**Orchestrator and version (e.g. Kubernetes, DC/OS, Swarm)**
Kubernetes

**What happened**:

**What you expected to happen**:

**How to reproduce it** (as minimally and precisely as possible):
For scale, which is a update scenario, we should merge with existing containerService properties. In this case, it is servicePrincipalProfile.Secret. The we do validation.

**Anything else we need to know**:

### PR DESCRIPTION
**Pull request by rjtsdl**
**What this PR does / why we need it**:

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #1269

**Special notes for your reviewer**:

**Release note**:

```release-note
```

---
This change is [Reviewable](https://reviewable.io/reviews/azure/acs-engine/1270)

### CODE REVIEW

**Comment by rjtsdl** [comment]
@seanknox , the package below is under BSD 3-clause. It is also used in acsrp. So if it is not registered, please help.
```
github.com/imdario/mergo
```

**Comment by jchauncey** [comment]
ginkgo tests failed with a known issue that should be fixed with #1292

File: glide.yaml [hunk 1]
```diff
@@ -50,9 +50,9 @@ import:
   version: a8983c062be4b565d723c478922d7736e04fdba4
 - package: github.com/Masterminds/semver
   version: v1.3.1
+- package: github.com/imdario/mergo
+  version: ^0.2.2
 testImport:
 - package: github.com/onsi/gomega
 - package: github.com/onsi/ginkgo
-  version: v1.3.1
```
Comment (jackfrancis) [inline glide.yaml]: we don't want to lose these version declarations

File: glide.yaml [hunk 2]
```diff
@@ -50,9 +50,9 @@ import:
...
 - package: github.com/kelseyhightower/envconfig
-  version: ~1.3.0
```
Comment (jackfrancis) [inline glide.yaml]: ditto

Comment (jackfrancis) [inline glide.yaml]: @JiangtianLi just add back those version strings so the diff is clean for these two packages, but don't worry about re-running glide or anything

File: glide.yaml (hunk 1 above)
Comment (rjtsdl) [inline glide.yaml]: Well, i just cannot avoid conflict with testImport with version specified.

Comment (jchauncey) [inline glide.yaml]: then you are running into the exact problem i was seeing earlier when i was trying to get glide to reconcile

Comment (rjtsdl) [inline glide.yaml]: How is your workaround? Mine was just remove the versions of testImport. @jchauncey

Comment (jchauncey) [inline glide.yaml]: i eventually stopped fiddling with it since it wasnt pressing.

File: glide.yaml (hunk 2 above)
Comment (rjtsdl) [inline glide.yaml]: So you are saying, i can keep glide.lock not synced with glide.yaml?

Comment (rjtsdl) [inline glide.yaml]: I guess, it is yes :)

**Pull request merged**